src/YAML.jl

```diff
@@ -97,8 +100,5 @@ load_file(filename::AbstractString, args...; kwargs...) =
     end
 
 load_all_file(filename::AbstractString, args...; kwargs...) =
-    open(filename, "r") do input
-        load_all(input, args...; kwargs...)
-    end
-
+    load_all(open(filename, "r"), args...; kwargs...)
```

Comment:
I'm fairly sure this won't close the file, in contrast to the `open(...) do` construction. Is that intentional?

Reply:
Yes. Since parsing is lazy, using `do` here results in the file being closed before anything is read.

Reply:
Ok, but it's still problematic to leave a file unclosed. It's entirely possible that's the cause of the Windows failures and even if it seems to work on the other platforms, it will be a file descriptor leak and if you read too many YAML files this way you will eventually fail because you have run out of file descriptors.

There are some options to handle this, with varying advantages and disadvantages.
1. Design the API with a manually called close operation.
2. Set up a finalizer in `YAMLDocIterator` which closes the file when the iterator is garbage collected.
3. Read the raw file eagerly and do the lazy parsing from an internal `IOBuffer`. That way the file can be closed immediately after reading.

Reply:
After looking closer at the code I think 3 is the better option. The main drawback is that it requires memory to store the raw file contents but if your file is large enough that that becomes a problem, you should use the `load_all(::IO)` method directly instead.

Reply:
Good observation that the `IOStream` object would have a finalizer. Unfortunately there are no guarantees that it will run before you run out of file descriptors, so better avoid relying on it.